apps/server/src/shared/repo/user/user.repo.ts

```diff
@@ -20,6 +20,11 @@ export class UserRepo {
 		return user;
 	}
 
+	async findByLdapId(ldapId: string): Promise<User> {
+		const user = await this.em.findOneOrFail(User, { ldapId });
+		return user;
+	}

```

Comment:
it is not good to assume that there is a one to one relation between ldapid and user.
It is possible that the same LDAP ID is used by two different LDAP systems for different users.
It is better to take the school into account in the relation.

Reply:
If it not fix with on this function it will be fail.
Users are not own by the system.
Users are own by the school at the moment.

Reply:
We have changed this, now we get all users with this ldapId and then filter by systemId, and thats how we make sure there is no duplicats, (uuids are unique for each system).

Reply:
implemented working solution with systemi.id and uuid